src/main/java/com/shopping_app/shoppingApp/repository/CartItemRepository.java

```diff
@@ -12,4 +15,9 @@ public interface CartItemRepository extends JpaRepository<CartItem, Long> {
     Optional<CartItem> findByIdAndCartId(Long cartItemId, Long cartId);
 
     Optional<CartItem> findByCartIdAndProductId(Long cartId, Long productId);
+
+    @Transactional
+    @Modifying
+    @Query("DELETE FROM CartItem n WHERE n.id = :id")
+    void deleteCartItems(Long id);
```

Comment:
Can this be not done with using a custom query?

Reply:
actually using default deleteById was not worked here , Reason i am adding cartitems from cart entity which is parent of cartItem entity. 
so due to this , when i try to delete cartItem directly using id ,  due to it being still attached and referenced by another entity it would be persisted again, hence the delete is cancelled.
so one of the solution fix is adding custom query. due to this i added .